### PR TITLE
feat: Finish hero section + app bar

### PR DESCRIPTION
**Pull request by doanhtu07**
# Status

- [x] Finish with UI
- [x] The app bar buttons will need to link to the sections or pages when they are finished
- [x] Where is the app bar on mobile view? Figma does not have it
- [x] Review https://github.com/acmutd/hackportal/pull/275
- [ ] The app bar icons are not finalized
- [ ] Still missing sign out option on app bar

**NOTE:** I've played around with the app bar scrolling effect. Let me know what you guys think.

# Showcase

## Navbar

https://github.com/acmutd/hackportal/assets/55064969/384f61e0-b539-4284-9222-f40ffdea708c

## Floating particle

If we don't want this, I can simply remove the transition. This floating effect is for comment from @uy-seng. Instead of scrolling and transitioning, I play around with the floating effect first.

We can discuss more about this.

https://github.com/acmutd/hackportal/assets/55064969/d211988a-f62b-4efc-9de7-d6c59ebe868c

### CODE REVIEW

**Comment by uy-seng** [comment]
we can link the position of circle relative to the scroll to animate the circle similar to figma, cuz in the figma the circle move to the right in each section of scroll

File: components/BackgroundCircles.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      //         duration: 0.4,
+      //       },
+      //     },
+      //   },
+
+      //   background: {
+      //     color: {
+      //       value: '#0d47a1',
+      //     },
+      //   },
```
Comment (vvyn) [inline components/BackgroundCircles.tsx]: You may want to delete this commented out code.

File: components/BackgroundCircles.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        // color: {
+        //   value: '#ffffff',
+        // },
+        // links: {
+        //   color: '#ffffff',
+        //   distance: 150,
+        //   enable: true,
+        //   opacity: 0.5,
+        //   width: 1,
+        // },
```
Comment (vvyn) [inline components/BackgroundCircles.tsx]: These as well.

File: components/homeComponents/HomeHero2.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+              className="w-full h-full object-cover"
+            />
+          </div>
+
+          {/* Big welcome */}
+          <div className="w-full flex flex-col gap-1 justify-center items-center bg-[rgba(242,243,255,0.75)] backdrop-blur-[70px]">
+            <p className="text-2xl md:text-4xl">Welcome To</p>
+            <h1 className="text-4xl md:text-6xl lg:text-9xl font-bold text-[#05149C]">
+              HACKPORTAL
+            </h1>
```
Comment (vvyn) [inline components/homeComponents/HomeHero2.tsx]: I think the font should be Fredoka One.

File: components/BackgroundCircles.tsx (hunk 1 above)
Comment (vvyn) [reply components/BackgroundCircles.tsx]: I liked the floating NavBar effect though.

**Review by vvyn — DISMISSED** [review]

File: components/AppHeader2/wrapper.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
+import { useEffect, useRef, useState } from 'react';
+
+import clsx from 'clsx';
+import AppHeader2_Core from './core';
+
+export const APP_HEADER_HEIGHT = 86;
+const INITIAL_HEADER_HEIGHT = APP_HEADER_HEIGHT;
+const TOP_OFFSET = 0; // Can be set to APP_HEADER_HEIGHT if want to app bar to be separated from the Hero part
+
+export default function AppHeader2_Wrapper() {
```
Comment (DedsecKnight) [inline components/AppHeader2/wrapper.tsx:10]: Thoughts on making the navbar sticky on top?

File: components/BackgroundCircles.tsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        },
+        size: {
+          value: Math.min(windowWidth * 0.75 * scaleFactor, 600 * scaleFactor),
+        },
+      },
+    }),
+    [windowWidth],
+  );
+
+  console.log(options, windowWidth);
```
Comment (DedsecKnight) [inline components/BackgroundCircles.tsx]: Remove this.

**Review by DedsecKnight — DISMISSED** [review]
Might wanna consider making the navbar sticky on top. Other than that, PR looks good to me and will merge it in once merge conflict are resolved.

File: components/AppHeader2/wrapper.tsx (hunk 4 above)
Comment (doanhtu07) [reply components/AppHeader2/wrapper.tsx:10]: Yeah I can do that. So no more hiding?

**Review by DedsecKnight — DISMISSED** [review]
lgtm! will merge this in once all the sections are linked properly.

**Review by vvyn — DISMISSED** [review]
looks good!

**Review by DedsecKnight — APPROVED** [review]
lgtm! will merge this in now since we already had 2 approvals prior.

**Pull request merged**